src/java_bytecode/expr2java.h

```diff
@@ -50,4 +51,34 @@ class expr2javat:public expr2ct
 std::string expr2java(const exprt &expr, const namespacet &ns);
 std::string type2java(const typet &type, const namespacet &ns);
 
+/// Convert floating point number to a string without printing
+/// unnecessary zeros. Prints decimal if precision is not lost.
+/// Prints hexadecimal otherwise, and/or java-friendly NaN and Infinity
+template <typename float_type>
+std::string floating_point_to_java_string(float_type value)
+{
+  static const std::string class_name = std::is_same<float_type, float>::value
```

Comment:
In the old code the following check was used:

```
 bool is_java_float=(src.type()==java_float_type());
```

I haven't completely understood what the above code would do, but is it not the case than on a platform where double is 32bits wide, `ieee_repr.is_double()` will return true for when the input has a width of 32bits (i.e. a Java float)

Reply:
IIRC the Java standard is quite restrictive about the link between float and double and the IEEE-754 binary32 and binary64.